cs411project/sql/check-machine-availability.sql

```diff
@@ -0,0 +1,83 @@
+/*
+ To test these queries, source heartbeat-insert-test.sql within the MySQL prompt
+  and then source this file. You should compare the output of this file versus
+  the expected results denoted in heartbeat-insert-test.sql. Note that the
+  test uses CURRENT_TIMESTAMP and these queries also user CURRENT_TIMESTAMP,
+  so it is important to run this query after inserting the test records because
+  if you wait too long (4 min), then all machines will be returned as 
+  AVAILABLE (since all heartbeats have timed out w.r.t. CURRENT_TIMESTAMP
+  of a later time) 
+*/
+
+
+
+/* Check the availability of a single machine (7 for this example) */
+/* Assuming for Machine.Status that 0 is BROKEN, 1 is ALIVE */
+SELECT
+  CASE WHEN Status = 0
+        THEN 'Machine is BROKEN' 
+       WHEN Status = 1 AND (
+          /* If the last heartbeat for a machine occurred AFTER (Tfail amount of time BEFORE the current time), then the machine is still in use */
+          SELECT SUBTIME(CURRENT_TIMESTAMP, Tfail) <= (CASE WHEN (LastTS IS NULL) THEN FirstTS ELSE LastTS END) AS StillInUse
+          FROM HeartbeatSequence
+          WHERE (LastTS = (SELECT CASE 
+                              /* Find the last timestamp between the FirstTS and LastTS columns out of all HeartbeatSequence rows */
+                              /* TODO: could be replaced with GREATEST() (MySQL, not in ANSI) */
```

Comment:
A TA on Piazza specified we have to use ANSI-SQL (an exception is made for string/date/time manipulation functions like SUBTIME